packages/core/src/test/notifications/controller.test.ts

```diff
@@ -107,6 +108,10 @@ describe('Notifications Controller', function () {
     })
 
     it('can fetch and store startup notifications', async function () {
+        // There seems to be a race condition with having a global spy object that is reset after each
+        // test. Doesn't seem to affect the other ones, for some reason.
+        const setContextSpy = sinon.spy(setContext, 'setContext')
```

Comment:
we could modify setContext.ts to store a map so that tests can assert it. That would avoid the need for spy/restore.

Reply:
Added